src/library/autodj/autodjprocessor.cpp

```diff
@@ -1127,12 +1144,28 @@ double AutoDJProcessor::getOutroStartSecond(DeckAttributes* pDeck) {
 
 double AutoDJProcessor::getOutroEndSecond(DeckAttributes* pDeck) {
     const mixxx::audio::FramePos trackEndPosition = pDeck->trackEndPosition();
+    const mixxx::audio::FramePos outroStartPosition = pDeck->outroStartPosition();
     const mixxx::audio::FramePos outroEndPosition = pDeck->outroEndPosition();
     if (!outroEndPosition.isValid() || outroEndPosition > trackEndPosition) {
-        return getLastSoundSecond(pDeck);
+        double lastSoundSecond = getLastSoundSecond(pDeck);
+        if (!outroStartPosition.isValid() || outroStartPosition > trackEndPosition) {
+            // No outro start and outro end set, use Last Sound.
+            return lastSoundSecond;
```

Comment:
wasn't this dependent on transition mode?

Reply:
Here we calculate a reliable OutroEnd position, as an input to the transition mode dependent code. 
The change here takes place in case we have an OutroStart. Than we try to use the transition time to calulate a better OutroEnd than just the lastSoundSecond as the last resort.